src/ralph/networks/models/networks.py

```diff
@@ -93,21 +94,65 @@ def next_free_hostname(self):
         """
         Retrieve next free hostname
         """
-        return AssetLastHostname.get_next_free_hostname(
-            self.hostname_template_prefix,
-            self.hostname_template_postfix,
-            self.hostname_template_counter_length
-        )
+        if self.use_hostname_counter:
+            return AssetLastHostname.get_next_free_hostname(
+                self.hostname_template_prefix,
+                self.hostname_template_postfix,
+                self.hostname_template_counter_length
+            )
+        return self.next_hostname_without_model_counter()
 
     def issue_next_free_hostname(self):
         """
         Retrieve and reserve next free hostname
         """
-        return AssetLastHostname.increment_hostname(
-            self.hostname_template_prefix,
-            self.hostname_template_postfix,
-        ).formatted_hostname(self.hostname_template_counter_length)
+        if self.use_hostname_counter:
+            return AssetLastHostname.increment_hostname(
+                self.hostname_template_prefix,
+                self.hostname_template_postfix,
+            ).formatted_hostname(self.hostname_template_counter_length)
+        return self.next_hostname_without_model_counter(True)
+
+    def next_hostname_without_model_counter(self, create_last_hostname=False):
+        """
+        Return hostname based on already added hostname
+
+        Args:
+            create_last_hostname: If create new AssetLastHostname
 
+        Returns:
+            hostname string
+        """
+        from ralph.data_center.models.physical import DataCenterAsset
+        from ralph.data_center.models.virtual import Cluster
+        from ralph.virtual.models import VirtualServer
+
+        hostname_models = [DataCenterAsset, VirtualServer, Cluster, IPAddress]
+        hostnames = []
+        for model_class in hostname_models:
+            item = model_class.objects.filter(
+                hostname__startswith=self.hostname_template_prefix,
+                hostname__endswith=self.hostname_template_postfix
+            ).order_by('-hostname').first()
+            if item and item.hostname:
+                hostnames.append(
+                    item.hostname[
+                        len(self.hostname_template_prefix):
+                        -len(self.hostname_template_postfix)
+                    ]
+                )
+        counter = 1
+        if hostnames:
+            hostnames = sorted(hostnames, reverse=True)
+            counter = int(hostnames[0]) + 1
+        hostname = AssetLastHostname(
+            prefix=self.hostname_template_prefix,
+            counter=counter,
+            postfix=self.hostname_template_postfix
+        )
+        if create_last_hostname:
+            hostname.save()
```

Comment:
we don't need to save (create?) `AssetLastHostname` when `use_hostname_counter` is `False`